Use stable features where possible

vmx opened the pull request:
`io::copy()` as well as the `BufReader`/`BufWriter` don't need nightly
Rust anymore as they the features needed are available on more recent
stable Rust versions. This commit bumps the MSRV to 1.55.

Closes #15.

vmx committed:
Use stable features where possible

`io::copy()` as well as the `BufReader`/`BufWriter` don't need nightly
Rust anymore as they the features needed are available on more recent
stable Rust versions. This commit bumps the MSRV to 1.55.

Closes #15.

Stebalien commented:
> Rust anymore as they the features needed are available on more recent
stable Rust versions

I don't think this is correct. The current code reveals uninitialized memory which is generally UB. The new method (still unstable) is to use https://doc.rust-lang.org/nightly/std/io/trait.Read.html#method.read_buf.

For now, I'd suggest just initializing a buffer unless nightly is available.

vmx committed:
Don't run doc tests

The doc tests currently don't compile, this is not a regression, it has
been the case even before the previous commit.

vmx commented:
I wanted to many thing at the same time with this PR. This should be split into separate ones:

 - Using a newer Rust version (MSRV) and check which features can now be implemented without requiring nightly
 - Upgrading nightly APIs to the latest nightly version
 - Implement `io::copy()` for stable Rust

Hence I'm closing this issue.
